src/sagemaker/tensorflow/estimator.py

```diff
@@ -10,230 +10,62 @@
-        """
-        name = tempfile.mkdtemp()
-        try:
-            yield name
-        finally:
-            shutil.rmtree(name)
-
-    def validate_requirements(self):
-        """Ensure that TensorBoard and the AWS CLI are installed.
-
-        These dependencies are required for using TensorBoard.
-
-        Raises:
-            EnvironmentError: If at least one requirement is not installed.
-        """
-        if not self._cmd_exists("tensorboard"):
-            raise EnvironmentError(
-                "TensorBoard is not installed in the system. Please install TensorBoard using the"
-                " following command: \n pip install tensorboard"
-            )
-
-        if not self._cmd_exists("aws"):
-            raise EnvironmentError(
-                "The AWS CLI is not installed in the system. Please install the AWS CLI using the"
-                " following command: \n pip install awscli"
-            )
-
-    def create_tensorboard_process(self):
-        """Create a TensorBoard process.
-
-        Returns:
-            tuple: A tuple containing:
-                int: The port number.
-                process: The TensorBoard process.
-
-        Raises:
-            OSError: If no ports between 6006 and 6105 are available for starting TensorBoard.
-        """
-        port = 6006
-
-        for _ in range(100):
-            p = subprocess.Popen(
-                [
-                    "tensorboard",
-                    "--logdir",
-                    self.logdir,
-                    "--host",
-                    "localhost",
-                    "--port",
-                    str(port),
-                ],
-                stdout=subprocess.PIPE,
-                stderr=subprocess.PIPE,
-            )
-            self.event.wait(5)
-            if p.poll():
-                port += 1
-            else:
-                return port, p
-
-        raise OSError(
-            "No available ports to start TensorBoard. Attempted all ports between 6006 and 6105"
-        )
-
-    def run(self):
-        """Run TensorBoard process."""
-        port, tensorboard_process = self.create_tensorboard_process()
-
-        logger.info("TensorBoard 0.1.7 at http://localhost:%s", port)
-        while not self.estimator.checkpoint_path:
-            self.event.wait(1)
-        with self._temporary_directory() as aws_sync_dir:
-            while not self.event.is_set():
-                args = ["aws", "s3", "sync", self.estimator.checkpoint_path, aws_sync_dir]
-                subprocess.call(args, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-                self._sync_directories(aws_sync_dir, self.logdir)
-                self.event.wait(10)
-        tensorboard_process.terminate()
-
-
 class TensorFlow(Framework):
     """Handle end-to-end training and deployment of user-provided TensorFlow code."""
 
     __framework_name__ = "tensorflow"
+    _SCRIPT_MODE_REPO_NAME = "tensorflow-scriptmode"
```

Comment:
Is this repo name used by the DLC images?

Reply:
no, it's used only by the TF 1.11-1.13 images. this PR still preserves the image URI stuff for older images, though arguably it could be removed as well. definitely open to it.

Reply:
Makes sense. I think it's probably safe to remove that functionality for older images and require users to specify the image URI directly, but I'm also fine with leaving it as is.

Reply:
I'll probably do it in a separate PR then, and add some appropriate warnings. maybe once #1464 is finished, it'll be cleaner.